src/operations/create_collection.ts

```diff
@@ -127,12 +133,15 @@ export class CreateCollectionOperation extends CommandOperation<Collection> {
         db.s.client.options.autoEncryption?.encryptedFieldsMap?.[`${db.databaseName}.${name}`];
 
       if (encryptedFields) {
+        // Creating a QE collection required min server of 7.0.0
+        if (server.description.maxWireVersion < MIN_SUPPORTED_QE_WIRE_VERSION) {
+          throw new MongoCompatibilityError(INVALID_QE_VERSION);
+        }
```

Comment:
Not at all a necessary change, but there is a class: `ServerCapabilities`: https://github.com/mongodb/node-mongodb-native/blob/main/src/sdam/topology.ts#L945-L989 That is _supposed_ to encode what the current wireVersion supports as a boolean method. 

I can find like 20 instances of this class not being used where it should be though, so maybe something we'd want to refactor as a whole later if at all.

Reply:
I've updated to mention the server version required in the error now. With respect to `ServerCapabilities` - I think it's better to refactor as a whole later on... It's better existing as a property on `Server` instead of `Topology` (hence the name and we can have mixed topologies) and the command execution at the operation layer of the codebase executes on a server, so it would be more readily accessible and relevant there. I've created https://jira.mongodb.org/browse/NODE-5253 for a spike to create the ticket.